cmplibrary/src/test/java/com/sourcepoint/cmplibrary/SourcePointClientTest.java

```diff
@@ -0,0 +1,205 @@
+
+import java.util.HashSet;
+
+import cz.msebera.android.httpclient.Header;
+
+import static android.os.Build.VERSION_CODES.O;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE, sdk = O)
+public class SourcePointClientTest {
+
+    AsyncHttpClient http;
+    SourcePointClient sourcePointClient;
+    ConsentLib.OnLoadComplete onLoadComplete;
+
+    @Before
+    public void setSourcePointClient() throws ConsentLibException.BuildException {
+        http = mock(AsyncHttpClient.class);
+        onLoadComplete = mock(ConsentLib.OnLoadComplete.class);
+
+        sourcePointClient = new SourcePointClientBuilder(808, "TEST", true).build();
+        sourcePointClient.setHttpDummy(http);
+        System.out.println("TTTT : 11 " + http);
+    }
+
+    JSONObject jsonObject = null;
+
+    private void doAPICallWithAnswer(final boolean onSuccess) {
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) {
+                System.out.println("TTTT : CALLED");
+                SourcePointClient.ResponseHandler listener = (SourcePointClient.ResponseHandler) invocation.getArguments()[1];
+
+                if (onSuccess) {
+                    String stringToParse = "{\"site_id\":\"http://google.com\"}";
+                    try {
+                        jsonObject = new JSONObject(stringToParse);
+                    } catch (JSONException err) {
+                        System.out.println("Error " + err.toString());
+                    }
+
+                    Header[] headers = new Header[]{};
+                    listener.onSuccess(200, headers, jsonObject);
+                } else {
+                    Header[] headers = new Header[]{};
+                    listener.onFailure(404, headers, "Error", new ConsentLibException("Error"));
+                }
+
+                return null;
+            }
+        }).when(http).get(anyString(), any(ResponseHandlerInterface.class));
+    }
+
+    @Test
+    public void getSiteID() {
+        doAPICallWithAnswer(true);
+
+        sourcePointClient.getSiteID(onLoadComplete);
+
+        verify(onLoadComplete, times(1)).onSuccess(eq("http://google.com"));
+
+        verify(onLoadComplete, never()).onFailure(any(ConsentLibException.class));
+
+        reset(onLoadComplete);
+
+        doAPICallWithAnswer(false);
+
+        sourcePointClient.getSiteID(onLoadComplete);
+
+        verify(onLoadComplete, never()).onSuccess(any());
+
+        verify(onLoadComplete, times(1)).onFailure(any(ConsentLibException.class));
+    }
```

Comment:
I'd break each of these test cases into two. One for success and one for failure. So, instead of 
```java
@Test
    public void getSiteID() 
```
We'd have;
```java
@Test
    public void getSiteIDOnSuccess() {}

@Test
    public void getSiteIDOnFailure() {}
```

Reply:
will refactor to one assertion per method